Creating an empty Pipeline

In [1]:
from quantopian.pipeline import Pipeline

def make_pipeline():
    
    return Pipeline()


In the above cell we have created an empty pipeline. This pipeline does not do anything unless we call the `run_pipeline` funtion on it. Lets do that now.

In [2]:
from quantopian.research import run_pipeline

In [5]:
# Give the dates
start = '2016-1-11'
end = '2016-1-11'

output = run_pipeline(make_pipeline(),
                     start,
                     end)

In [6]:
output.head()

Empty DataFrame
Columns: []
Index: [(2016-01-11 00:00:00+00:00, Equity(2 [ARNC])), (2016-01-11 00:00:00+00:00, Equity(21 [AAME])), (2016-01-11 00:00:00+00:00, Equity(24 [AAPL])), (2016-01-11 00:00:00+00:00, Equity(25 [ARNC_PR])), (2016-01-11 00:00:00+00:00, Equity(31 [ABAX]))]

We have just run an empty pipeline, that returns an empty dataframe with the stock symbols. We can count the list of symbols.

In [11]:
symbol_list = output.index.levels[1].unique().tolist()
len(symbol_list)

8401

The empty dataframe has 8401 symbols.

Next we will learn how to get data into the pipeline.